packages/react-components/priority-overflow/src/overflowManager.ts

```diff
@@ -33,6 +36,60 @@ export function createOverflowManager(): OverflowManager {
     update();
   });
 
+  function updateGroupVisibility(groupId: string) {
+    const group = groups[groupId];
+    if (group.invisibleItemIds.size && group.visibleItemIds.size) {
+      groupVisibility[groupId] = 'overflow';
+    } else if (group.visibleItemIds.size === 0) {
+      groupVisibility[groupId] = 'hidden';
+    } else {
+      groupVisibility[groupId] = 'visible';
+    }
+  }
+
+  const getNextItem = (queueToDequeue: PriorityQueue<string>, queueToEnqueue: PriorityQueue<string>) => {
+    const nextItem = queueToDequeue.dequeue();
+    queueToEnqueue.enqueue(nextItem);
+    return overflowItems[nextItem];
+  };
+
+  const groupManager = {
+    isGroupVisible(groupId: string) {
+      return groupVisibility[groupId] === 'visible' || groupVisibility[groupId] === 'overflow';
+    },
+    isSingleVisible(itemId: string, groupId: string) {
+      return (
+        this.isGroupVisible(groupId) &&
+        groups[groupId].visibleItemIds.has(itemId) &&
+        groups[groupId].visibleItemIds.size === 1
+      );
+    },
```

Comment:
```suggestion
    isItemVisible(itemId: string, groupId: string) {
      return groups[groupId].visibleItemIds.has(itemId);
    },
```

The other check's aren't really necessary right?